DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Rules/AnonymizerRuleFactory.cs

```diff
@@ -0,0 +1,177 @@
+                }
+
+                ruleSetting = _customizedSettings[ruleContent[Constants.RuleSetting].ToString()];
+            }
+
+            if (parameters != null)
+            {
+                if (ruleSetting == null)
+                {
+                    ruleSetting = parameters;
+                }
+                else
+                {
+                    ruleSetting.Merge(parameters, new JsonMergeSettings { MergeArrayHandling = MergeArrayHandling.Concat });
+                }
+            }
+
+            // Parse and validate tag
+            if (ruleContent.ContainsKey(Constants.TagKey))
+            {
+                var tagContent = ruleContent[Constants.TagKey].ToString();
+                if (TryParseDICOMTag(tagContent, out DicomTag tag))
+                {
+                    return new AnonymizerTagRule(tag, method, ruleSetting, ruleContent.ToString(), _processorFactory, _settingsFactory);
+                }
+                else if (TryParseDICOMMaskedTag(tagContent, out DicomMaskedTag maskedTag))
+                {
+                    return new AnonymizerMaskedTagRule(maskedTag, method, ruleSetting, ruleContent.ToString(), _processorFactory, _settingsFactory);
+                }
+                else if (TryParseDICOMVR(tagContent, out DicomVR vr))
+                {
+                    return new AnonymizerVRRule(vr, method, ruleSetting, ruleContent.ToString(), _processorFactory, _settingsFactory);
+                }
+                else if (TryParseDICOMTagName(tagContent, out DicomTag tagByName))
+                {
+                    return new AnonymizerTagRule(tagByName, method, ruleSetting, ruleContent.ToString(), _processorFactory, _settingsFactory);
+                }
+                else
+                {
+                    throw new AnonymizationConfigurationException(DicomAnonymizationErrorCode.InvalidConfigurationValues, "Invaid tag in rule config");
+                }
+            }
+            else
+            {
+                throw new AnonymizationConfigurationException(DicomAnonymizationErrorCode.MissingConfigurationFields, "Missing tag in rule config");
+            }
+        }
+
+        private bool TryParseDICOMTag(string tagContent, out DicomTag output)
+        {
+            try
+            {
+                output = DicomTag.Parse(tagContent);
+                return true;
+            }
+            catch
+            {
+                output = null;
+                return false;
+            }
+        }
+
+        private bool TryParseDICOMMaskedTag(string tagContent, out DicomMaskedTag output)
+        {
+            try
+            {
+                output = DicomMaskedTag.Parse(tagContent);
+                return true;
+            }
+            catch
+            {
+                output = null;
+                return false;
+            }
```

Comment:
I would like to write code like this:

output = null;
result = false;
try {
   output = XXX.parse(...);
   result = true
}
catch {}

return result;

Reply:
Both are fine for me.